realm/realm-library/src/androidTest/java/io/realm/HandoverTests.java

```diff
@@ -0,0 +1,149 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.test.AndroidTestCase;
+import android.util.Log;
+
+import io.realm.entities.AllTypes;
+import io.realm.entities.Cat;
+import io.realm.entities.Dog;
+import io.realm.entities.Owner;
+import io.realm.util.RealmBackgroundTask;
+
+/**
+ * Class for testing the public handover functionality
+ */
+public class HandoverTests extends AndroidTestCase {
+
+    private static final int TEST_DATA_SIZE = 10;
+    private RealmConfiguration config;
+    private Realm realm;
+
+    @Override
+    protected void setUp() throws Exception {
+        config = TestHelper.createConfiguration(getContext());
+        Realm.deleteRealm(config);
+        realm = Realm.getInstance(config);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        if (realm != null) {
+            realm.close();
+        }
+    }
+
+    // Add a number of AllType objects with an increasing number of items in `columnRealmList`
+    private void populateTestRealm(int dataSize) {
+        realm.beginTransaction();
+        for (int i = 0; i < dataSize; i++) {
+            Owner owner = realm.createObject(Owner.class);
+            owner.setName("Owner " + i);
+
+            Cat cat = realm.createObject(Cat.class);
+            cat.setName("Cat " + i);
+            owner.setCat(cat);
+
+            // Add increasing number of dogs
+            for (int j = 0; j < i; j++) {
+                Dog dog = realm.createObject(Dog.class);
+                dog.setName(String.format("Dog (%d, %d)", i, j));
+                dog.setAge(j);
+                dog.setOwner(owner);
+                owner.getDogs().add(dog);
+            }
+        }
+        realm.commitTransaction();
+    }
+
+    // Query on a regular table
+    public void testThreadLocalVersion_realmQuery_tableView() {
+        populateTestRealm(TEST_DATA_SIZE);
+
+        final RealmQuery<Owner> query = realm.where(Owner.class);
+        RealmBackgroundTask bgTask = new RealmBackgroundTask(config);
+        bgTask.run(new RealmBackgroundTask.Task() {
+            @Override
+            public void run(Realm realm) {
+                RealmQuery<Owner> bgQuery = realm.threadLocalVersion(query);
+                RealmResults<Owner> results = bgQuery.findAll();
+                assertEquals(TEST_DATA_SIZE, results.size());
+            }
+        });
+        bgTask.checkFailure();
+    }
+
+    // Query on a table view (sub query)
+    public void testThreadLocalVersion_realmQuery_tableViewSubQuery() {
+        populateTestRealm(10);
```

Comment:
Because the following tests make assumptions about the size that would reduce readability quite a lot if converted to things lime (TEST_DATA_SIZE - 1), also it require at least 2 elements.